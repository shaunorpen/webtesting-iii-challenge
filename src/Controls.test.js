import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Controls from './Controls';

afterEach(rtl.cleanup);

let wrapper;

describe('Controls component:', () => {
    it('renders correctly', () => {
      wrapper = rtl.render(<Controls />);
      expect(wrapper.container).toMatchSnapshot();
    });

    it('provides buttons to toggle the closed and locked states.', () => {
      wrapper = rtl.render(<Controls />);
      expect(wrapper.getByText(/Close Gate/)).toBeInTheDocument();
      expect(wrapper.getByText(/Lock Gate/)).toBeInTheDocument();
    });

    it('button text changes to reflect the state the door will be in if clicked', () => {
      wrapper = rtl.render(<Controls />);
    });

    it('the closed toggle button is disabled if the gate is locked', () => {
      wrapper = rtl.render(<Controls locked={true}/>);
      expect(wrapper.getByText(/Close Gate/)).toBeDisabled();
    });

    it('the locked toggle button is disabled if the gate is open', () => {
      wrapper = rtl.render(<Controls open={true}/>);
      expect(wrapper.getByText(/Lock Gate/)).toBeDisabled();
    });
});