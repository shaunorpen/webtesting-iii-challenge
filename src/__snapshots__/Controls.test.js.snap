// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component: renders correctly 1`] = `
<div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;
